ci/deps/actions-39.yaml

```diff
@@ -36,7 +36,7 @@ dependencies:
   - psycopg2
   - pymysql
   - pytables
-  - pyarrow=5
```

Comment:
i think @jorisvandenbossche  did this recently-ish to make sure we had coverage on various pyarrow versions

Reply:
I've been trying to avoid pins in these files such that our testing becomes more aligned https://github.com/pandas-dev/pandas/issues/29685

I made https://github.com/pandas-dev/pandas/issues/45705 to address the coverage via a build matrix instead